### ISSUE TITLE
False positive dangling diagnostics when the [[gsl::Owner]] and [[clang::lifetimebound]] attributes are used together

### ISSUE THREAD

**Issue by hokein** [llvm/llvm-project#108272]
Clang emits bogus dangling diagnostics when the `[[gsl::Owner]]` and `[[liftimebound]]` attributes are used together.

Consider the following case https://godbolt.org/z/z6fdqGEa7:

```
template <typename T>
struct [[gsl::Owner]] StatusOr {
  const T &value() [[clang::lifetimebound]];
};

template <typename V>
class Wrapper {
 public:
  operator V() const { return value; } 
  V value;
};

std::string_view test() {
  StatusOr<Wrapper<std::string_view>> k;
  std::string_view good = StatusOr<Wrapper<std::string_view>>().value(); // bad1: emit dangling diagnostic

  return k.value(); // bad2: emit dangling diagnostic
}
```

We should not emit diagnostics for these two cases.  (Removing either the `[[gsl::Owner]]` or `[[clang::lifetimebound]]` suppresses the diagnostics).
